src/v/storage/disk_log_impl.cc

```diff
@@ -2343,22 +2333,6 @@ disk_log_impl::get_reclaimable_offsets(gc_config cfg) {
         co_return res;
     }
 
-    /*
-     * there is currently a bug with read replicas that makes the max
-     * collectible offset unreliable. the read replica topics still have a
-     * retention setting, but we are going to exempt them from forced reclaim
-     * until this bug is fixed to avoid any complications.
-     *
-     * https://github.com/redpanda-data/redpanda/issues/11936
-     */
-    if (config().is_read_replica_mode_enabled()) {
```

Comment:
I think you'll need to remove a similar check from the code that Noah just merged. See [here](https://github.com/redpanda-data/redpanda/pull/12091/files#diff-8caa0c9719de25bd209fa674959d9327cd55237de2fb4683761acf9bcc6de217R2063).

Reply:
Done